tools/system_libs.py

```diff
@@ -926,11 +926,23 @@ def retrieve():
       logger.warning('retrieving port: ' + name + ' from ' + url)
       try:
         from urllib.request import urlopen
+        f = urlopen(url)
+        data = f.read()
       except ImportError:
         # Python 2 compatibility
-        from urllib2 import urlopen
-      f = urlopen(url)
-      data = f.read()
+        from urllib2 import urlopen, URLError
+        try:
+          f = urlopen(url)
+          data = f.read()
+        except URLError as e:
+          try:
+            import requests
```

Comment:
may be worth a comment that `requests` is used as a workaround for an older python?